kratos/processes/from_json_check_result_process.cpp

```diff
@@ -0,0 +1,866 @@
+                        components_vector[i_vector][i_step] = r_vector[i_vector];
+                    }
+                }
+                for (IndexType i_vector = 0; i_vector < size_vector; ++i_vector) {
+                    r_var_database.SetValues(r_time,components_vector[i_vector], i, i_vector, i_gp);
+                }
+            }
+        }
+    }
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void FromJSONCheckResultProcess::FillVariablesList(
+    const std::vector<std::string>& rNodalVariablesNames,
+    const std::vector<std::string>& rGPVariablesNames
+    )
+{
+    KRATOS_TRY;
+
+    // Nodal variables
+    for (const std::string& r_variable_name : rNodalVariablesNames) {
+        if (KratosComponents<Variable<double>>::Has(r_variable_name)){
+            mpNodalVariableDoubleList.push_back(&(KratosComponents< Variable<double>>::Get(r_variable_name)));
+        } else if (KratosComponents<Variable<array_1d<double, 3> >>::Has(r_variable_name)) {
+            mpNodalVariableArrayList.push_back(&(KratosComponents<Variable<array_1d<double, 3>>>::Get(r_variable_name)));
+        } else if (KratosComponents<Variable<Vector>>::Has(r_variable_name)) {
+            mpNodalVariableVectorList.push_back(&(KratosComponents<Variable<Vector>>::Get(r_variable_name)));
+        } else {
+            KRATOS_ERROR << "Only double, array and vector are allowed in the variables list. Variable: " << r_variable_name << std::endl;
+        }
+    }
+
+    // GP variables
+    for (const std::string& r_variable_name : rGPVariablesNames) {
+        if (KratosComponents<Variable<double>>::Has(r_variable_name)){
+            mpGPVariableDoubleList.push_back(&(KratosComponents< Variable<double>>::Get(r_variable_name)));
+        } else if (KratosComponents<Variable<array_1d<double, 3> >>::Has(r_variable_name)) {
+            mpGPVariableArrayList.push_back(&(KratosComponents<Variable<array_1d<double, 3>>>::Get(r_variable_name)));
+        } else if (KratosComponents<Variable<Vector>>::Has(r_variable_name)) {
+            mpGPVariableVectorList.push_back(&(KratosComponents<Variable<Vector>>::Get(r_variable_name)));
+        } else {
+            KRATOS_ERROR << "Only double, array and vector variables are allowed in the variables list. Variable: " << r_variable_name << std::endl;
+        }
+    }
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+bool FromJSONCheckResultProcess::CheckValues(
+    const double ValueEntity,
+    const double ValueJSON
+    )
+{
+    const bool check = std::abs(ValueEntity-ValueJSON) <= std::max(mRelativeTolerance * std::max(std::abs(ValueEntity), std::abs(ValueJSON)), mAbsoluteTolerance) ? true : false;
```

Comment:
This should take the ValueJSON only, not the max between the ValueEntity and it, since the latter could be rubbish.

Reply:
It is done in the same way as in the python version, if I change behaviour now I will broke tests, this should be updated in a future PR.